packages/db-schema/src/repositories/admin.ts

```diff
@@ -0,0 +1,300 @@
+import { and, count as drizzleCount, eq, ilike } from "drizzle-orm";
+import { Logger } from "pino";
+
+import { admins } from "../core/defs.js";
+import { InsertAdmin, SelectAdmin } from "../core/types.js";
+import { Database } from "../types.js";
+import { SearchAdminsParams } from "./types/index.js";
+import { PartialExcept } from "./util/types.js";
+
+/**
+ * Admin Repository
+ * Handles database operations for admins
+ */
+export class AdminRepository {
```

Comment:
Ok this is a repository. All the repositories in this folder follow this same pattern.

A repository is now modeled as a class. I went back and forth on whether to use classes or plain objects (I don't like classes personally), and went with class because I've seen other people on the team prefer them. Encapsulating repository functions in a class gives us a nice way to pass in any dependency the repository has. Usually this is a database instance and a logger.